src/main/java/com/redhat/devtools/intellij/qute/lang/format/QuteFormattingModelBuilder.java

```diff
@@ -0,0 +1,354 @@
+                                                           @Nullable Alignment alignment,
+                                                           @Nullable List<DataLanguageBlockWrapper> foreignChildren,
+                                                           @NotNull CodeStyleSettings codeStyleSettings) {
+    final FormattingDocumentModelImpl documentModel = FormattingDocumentModelImpl.createOn(node.getPsi().getContainingFile());
+    HtmlPolicy policy = new HtmlPolicy(codeStyleSettings, documentModel);
+    /* XXX return HbTokenTypes.TAGS.contains(node.getElementType()) ?
+           new HandlebarsTagBlock(node, wrap, alignment, this, codeStyleSettings, foreignChildren, policy) :
+           new HandlebarsBlock(node, wrap, alignment, this, codeStyleSettings, foreignChildren, policy);*/
+    return new HandlebarsBlock(node, wrap, alignment, this, codeStyleSettings, foreignChildren, policy);
+  }
+
+  /**
+   * We have to override {@link TemplateLanguageFormattingModelBuilder#createModel}
+   * since after we delegate to some templated languages, those languages (xml/html for sure, potentially others)
+   * delegate right back to us to format the HbTokenTypes.OUTER_ELEMENT_TYPE token we tell them to ignore,
+   * causing a stack-overflowing loop of polite format-delegation.
+   */
+  @Override
+  public @NotNull FormattingModel createModel(@NotNull FormattingContext formattingContext) {
+    /*if (!HbConfig.isFormattingEnabled()) {
+      // formatting is disabled, return the no-op formatter (note that this still delegates formatting
+      // to the templated language, which lets the users manage that separately)
+      return new SimpleTemplateLanguageFormattingModelBuilder().createModel(formattingContext);
+    }
+*/
+    final PsiFile file = formattingContext.getContainingFile();
+    Block rootBlock;
+
+    ASTNode node = formattingContext.getNode();
+
+    if (node.getElementType() == QuteElementTypes.QUTE_OUTER_ELEMENT_TYPE) {
+      // If we're looking at a HbTokenTypes.OUTER_ELEMENT_TYPE element, then we've been invoked by our templated
+      // language.  Make a dummy block to allow that formatter to continue
+      return new SimpleTemplateLanguageFormattingModelBuilder().createModel(formattingContext);
+    }
+    else {
+      rootBlock = getRootBlock(file, file.getViewProvider(), formattingContext.getCodeStyleSettings());
+    }
+    return new DocumentBasedFormattingModel(
+      rootBlock, formattingContext.getProject(), formattingContext.getCodeStyleSettings(), file.getFileType(), file);
+  }
+
+  /**
+   * Do format my model!
+   *
+   * @return false all the time to tell the {@link TemplateLanguageFormattingModelBuilder}
+   * to not-not format our model (i.e. yes please!  Format away!)
+   */
+  @Override
+  public boolean dontFormatMyModel() {
+    return false;
+  }
+
+  private static class HandlebarsTagBlock extends HandlebarsBlock {

```

Comment:
Replace Handlebars with Qute everywhere

Reply:
Ok I will do that, just for your information, I keep all handlebars code which manages auto indentation for Handlebars block. In a separate PR I will try to do the same thing for Qute section.

Reply:
fixed